ph_rando/patcher/_util.py

```diff
@@ -397,6 +398,69 @@ def _patch_bmg_events(areas: list[Area], input_rom: rom.NintendoDSRom) -> None:
             )
 
 
+def _patch_system_bmg(input_rom: rom.NintendoDSRom) -> None:
+    """
+    Patches the system BMG file to include all non-global "got item" BMG messages,
+    effectively making them global.
+    The "got item" message is the text that pops up when Link collects an item.
+    """
+    gims = list()
+
+    bmg_data = bmg.BMG(input_rom.getFileByName(BMGS['MainIsland']))
+    # Progressive Sword GIM
+    gims.append(bmg_data.messages[0x0].stringParts)
+    bmg_data = bmg.BMG(input_rom.getFileByName(BMGS['Brave']))
+    # Bow GIM
+    gims.append(bmg_data.messages[0x5F].stringParts)
+    # Shovel GIM
+    gims.append(bmg_data.messages[0x60].stringParts)
+    bmg_data = bmg.BMG(input_rom.getFileByName(BMGS['Flame']))
+    # Boomerang GIM
+    gims.append(bmg_data.messages[0xDC].stringParts)
+    bmg_data = bmg.BMG(input_rom.getFileByName(BMGS['Kaitei']))
+    # SW/NW/SE/NE Sea Charts GIM
+    gims.append(bmg_data.messages[0x1].stringParts)
+    gims.append(bmg_data.messages[0x2].stringParts)
+    gims.append(bmg_data.messages[0x3].stringParts)
+    gims.append(bmg_data.messages[0x4].stringParts)
+    bmg_data = bmg.BMG(input_rom.getFileByName(BMGS['WisdomDungeon']))
+    # Hammer GIM
+    gims.append(bmg_data.messages[0x0].stringParts)
+    # King Key GIM
+    gims.append(bmg_data.messages[0x1].stringParts)
+    bmg_data = bmg.BMG(input_rom.getFileByName(BMGS['Frost']))
+    # Grappling Hook GIM
+    gims.append(bmg_data.messages[0xDB].stringParts)
+    bmg_data = bmg.BMG(input_rom.getFileByName(BMGS['Hidari']))
+    # Fishing Rod GIM
+    gims.append(bmg_data.messages[0x64].stringParts)
+    bmg_data = bmg.BMG(input_rom.getFileByName(BMGS['Torii']))
+    # Cannon GIM
+    gims.append(bmg_data.messages[0x0].stringParts)
+    # Salvage Arm GIM
+    gims.append(bmg_data.messages[0x1].stringParts)
+    bmg_data = bmg.BMG(input_rom.getFileByName(BMGS['Sea']))
+    # Sun Key GIM
+    gims.append(bmg_data.messages[0x1B].stringParts)
+    bmg_data = bmg.BMG(input_rom.getFileByName(BMGS['Ghost']))
+    # Ghost Key GIM
+    gims.append(bmg_data.messages[0x3E].stringParts)
+    bmg_data = bmg.BMG(input_rom.getFileByName(BMGS['Myou']))
+    # Regal Necklace GIM
+    gims.append(bmg_data.messages[0x0].stringParts)
+
+    # To avoid overwriting non-empty entries
+    if len(gims) > 0xA9 - 0x8A:
+        logger.warning(f'Aborting GIM patching, overflow ({len(gims)}/{0xa9 - 0x8a}).')
+        return
```

Comment:
We use `logger.warning` in a lot of places, so I would throw an exception here so it doesn't fail silently. Plus using an exception will cause a CI failure if it does overflow.

```suggestion
    if len(gims) > 0xA9 - 0x8A:
        raise Exception(f'Aborting GIM patching, overflow ({len(gims)}/{0xa9 - 0x8a}).')
        return
```